### ISSUE TITLE
Base context file will show wrong number of pulses for SA1/3

### ISSUE THREAD

**Issue by philsmt** [European-XFEL/DAMNIT#83]
https://github.com/European-XFEL/DAMNIT/blob/c385fc1c900b28b48e92936f9edd5934a04a0e79/damnit/base_context_file.py#L49

By chance I noticed that the base context file uses the `numberOfBunchesActual` property, which contains the sum of SA1/SA3 pulses in those respective tunnels:

![image](https://github.com/European-XFEL/DAMNIT/assets/56224768/4b1c07d6-f756-43b9-a5c9-d02002f1141c)

This may be confusing to someone starting with the base context file at these SASEs. If you want to continue using the XGM as data source here, one could try to infer the topic you're running in from the XGM's device ID.

**Comment by JamesWrigley** [comment]
I wonder if we could shave this yak another way by using the XGM component in the context file, and getting the component to infer the topic it's in :thinking:

**Comment by philsmt** [comment]
Yes, definitely. As it happens I already got plenty of topic/instrument/sase-inferring code in the pulse components, which I meant to abstract away into some utils anyway together with device-locating code.

**Comment by JamesWrigley** [comment]
This has been fixed for a while now that we have per-SASE templates: https://github.com/European-XFEL/DAMNIT/blob/master/damnit/ctx-templates/SA3_base.py
